p2p/src/network/pnet/p2p_network_pnet_reducer.rs

```diff
@@ -34,21 +40,92 @@ impl Half {
 }
 
 impl P2pNetworkPnetState {
-    pub fn reducer(&mut self, action: redux::ActionWithMeta<&P2pNetworkPnetAction>) {
-        let (action, meta) = action.split();
+    /// Substate is accessed
+    pub fn reducer<Action, State>(
+        mut state_context: Substate<Action, State, P2pNetworkSchedulerState>,
+        action: redux::ActionWithMeta<&P2pNetworkPnetAction>,
+    ) -> Result<(), String>
+    where
+        State: crate::P2pStateTrait,
+        Action: crate::P2pActionTrait<State>,
+    {
+        let (action, _meta) = action.split();
+        let pnet_state = &mut state_context
+            .get_substate_mut()?
+            .connection_state_mut(action.addr())
+            .ok_or_else(|| format!("Missing connection for action: {:?}", action))?
+            .pnet;
 
         match action {
-            P2pNetworkPnetAction::IncomingData { data, .. } => {
-                self.incoming.reduce(&self.shared_secret, data);
+            P2pNetworkPnetAction::IncomingData { data, addr } => {
+                pnet_state.incoming.reduce(&pnet_state.shared_secret, data);
+
+                let (dispatcher, state) = state_context.into_dispatcher_and_state();
+                let scheduler_state: &P2pNetworkSchedulerState = state.substate()?;
+                let pnet_state = &scheduler_state
+                    .connection_state(addr)
+                    .ok_or_else(|| format!("Missing connection for action: {:?}", action))?
+                    .pnet;
+
+                let Half::Done { to_send, .. } = &pnet_state.incoming else {
+                    return Ok(());
```

Comment:
Can we clarify under what conditions we can reach other states here?
Should this be filtered out by an enabling condition?

Reply:
If incoming data is small other state can be reached, it is not bug and it is normal for other incoming to be in `Half::Buffering`

Reply:
Please add a small comment mentioning that